src/eager.jl

```diff
@@ -0,0 +1,50 @@
+mutable struct EagerContext
+    ptr::Ptr{Cvoid}
+
+    function EagerContext()
+        options = @tfcall(:TFE_NewContextOptions, Ptr{Cvoid}, ())
+        status = Status()
+        context = @tfcall(:TFE_NewContext, Ptr{Cvoid}, (Ptr{Cvoid}, Ptr{Cvoid}), options, status)
+        check_status(status)
+        this = new(context)
+        finalizer(this) do self
+            @tfcall(:TFE_DeleteContext, Cvoid, (Ptr{Cvoid},), self.ptr)
+        end
+        @tfcall(:TFE_DeleteContextOptions, Cvoid, (Ptr{Cvoid},), options)
+        return this
+    end
+end
+
+
+mutable struct TensorHandle
```

Comment:
this doesn't need to be mutable does it?

Reply:
I think it does so that it can have a finalizer.